File: SchoolWebApp-Client/src/app/cbe/exams/models/school-exam.ts
import {ResourceModel} from '@/core/models/ResourceModel';

/**
 * The exam "event" header (type, term, schedule + release workflow).
 * The per-class-per-subject detail lives on {@link Exam}, which carries a
 * schoolExamId pointing back here.
 */
export class SchoolExam extends ResourceModel<SchoolExam> {
    public examStartDate?: string;
    public examEndDate?: string;
    public examMarkEntryEndDate?: string;
    public description?: string;
    public examTypeId?: number;
    public examType?: any;
    public sessionId?: number;
    public session?: any;

    // Release workflow
    public isReleased?: boolean;
    public releasedBy?: string;
    public releasedDate?: string;
    public parentsNotified?: boolean;
    public parentsNotifiedDate?: string;

    constructor(model?: Partial<SchoolExam>) {
        super(model);
    }
}

File: SchoolWebApp-API/Project.Core/Entities/CBE/Exams/SchoolExam.cs
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.CBE.Exams
{
    /// <summary>
    /// The exam "event" header - e.g. "Opening Exam, Term 2 2026, CBE".
    /// Holds the schedule (start/end/mark-entry dates) and the release
    /// workflow once, so a deadline can be changed in one place and the
    /// whole exam released at once. The per-class-per-subject detail
    /// (mark, results) lives on the child <see cref="Exam"/> rows.
    /// The Session FK gives term + academic year + curriculum.
    /// </summary>
    public class SchoolExam : Base
    {
        public DateOnly ExamStartDate { get; set; }
        public DateOnly ExamEndDate { get; set; }
        public DateOnly ExamMarkEntryEndDate { get; set; }

        public string? Description { get; set; }

        public int ExamTypeId { get; set; }
        public ExamType? ExamType { get; set; }
        public int SessionId { get; set; }
        public Session? Session { get; set; }

        // Release workflow. Releasing publishes results (e.g. to the
        // dashboard summary) and is the hook for parent notifications.
        public bool IsReleased { get; set; }
        public string? ReleasedBy { get; set; }
        public DateTime? ReleasedDate { get; set; }

        // Parent-notification stubs - persisted now, wired to delivery later.
        public bool ParentsNotified { get; set; }
        public DateTime? ParentsNotifiedDate { get; set; }

        public List<Exam> Exams { get; set; } = new();
    }
}
